pkg/client/restclient/request.go

```diff
@@ -517,6 +517,20 @@ func (r *Request) Timeout(d time.Duration) *Request {
 	return r
 }
 
+func (r *Request) encoder() (runtime.Encoder, bool) {
+	negotiated := r.content.NegotiatedSerializer
```

Comment:
@smarterclayton - I'm still trying to have this working and I still didn't think about performance of that solution. I will get back to it, once I have tests passing.